collector/lib/rust/host/src/kernel.rs

```diff
@@ -0,0 +1,93 @@
+use regex::Regex;
+use uname::uname;
+
+#[derive(Clone)]
+pub struct KernelVersion {
+    pub kernel: u64,
+    pub major: u64,
+    pub minor: u64,
+    pub build_id: u64,
+
+    pub release: String,
+
+    #[allow(dead_code)]
+    pub version: String,
+    #[allow(dead_code)]
+    pub machine: String,
+}
+
+impl Default for KernelVersion {
+    fn default() -> Self {
+        KernelVersion {
+            kernel: 0,
+            major: 0,
+            minor: 0,
+            build_id: 0,
+            release: "".to_string(),
+            version: "".to_string(),
+            machine: "".to_string(),
+        }
+    }
+}
+
+impl KernelVersion {
+    pub fn new(release: &str, version: &str, machine: &str) -> Self {
```

Comment:
Do we need these to be &str? Can't we use straight up String? Most places it looks like we already have those, it would save creating/copying strings.